src/libraries/Common/tests/System/IO/TempDirectory.cs

```diff
@@ -42,5 +46,29 @@ private void DeleteDirectory()
             try { Directory.Delete(Path, recursive: true); }
             catch { /* Ignore exceptions on disposal paths */ }
         }
+
+        /// <summary>
+        /// Generates a string with 255 random valid filename characters.
+        /// 255 is the max file/folder name length in NTFS and FAT32:
+        // https://docs.microsoft.com/en-us/windows/win32/fileio/filesystem-functionality-comparison?redirectedfrom=MSDN#limits
+        /// </summary>
+        /// <returns>A 255 length string with random valid filename characters.</returns>
+        public static string GetMaxLengthRandomName()
+        {
+            string guid = Guid.NewGuid().ToString("N");
+            StringBuilder sb = new StringBuilder(MaxNameLength, MaxNameLength);
+            while (sb.Length < MaxNameLength)
```

Comment:
I think a while loop appending a substring (where simple math determines whether substring length is not the whole string  ) would be more elegant than throw/catch.

Or just `guid + new string('x', 255 - guid.Length)

Reply:
Ok sure no problem. Let me make the change.
I also had some CI failures, I'm taking a look in case they are related to my last commits.